File: Python/DLLInsertionAtBeginning.py
class Node:
    def __init__(self, data):
        self.data = data
        self.prev = None
        self.next = None
class DLLinsertionAtBeginning:
    
    def __init__(self):
        self.head = None
    
    def insertionAtBeginning(self, newNode):
        newNode = Node(newNode)
        if self.head == None:
            self.head = newNode
        else:
            newNode.next = self.head
            self.head.prev = newNode
            self.head = newNode
    def displayDoublyLinkedList(self):
        current = self.head
        if(current!=None):
            print("Linked list contains: ", end=" ")
            while(current!=None):
                print(current.data, end=" ")
                current = current.next
            print()
        else:
            print("Linked list is empty")

dll = DLLinsertionAtBeginning()
dll.insertionAtBeginning(400)
dll.insertionAtBeginning(300)
dll.insertionAtBeginning(200)
dll.insertionAtBeginning(100)
dll.displayDoublyLinkedList()

File: Java/DLLInsertionAtBeginning.java
public class DLLInsertionAtBeginning{
    class Node{
        int data;
        Node prev;
        Node next;
        Node(int data){
            this.data = data;
            this.prev = null;
            this.next=null;
        }
    }
    Node head;
    public void insertAtBeginningDLL(int data){
        Node newNode = new Node(data);
        if (head==null){
            head = newNode;
        }else{
            newNode.next = head;
            head.prev = newNode;
            head = newNode;
        }
    }
    public void display(){
        Node current = head;
        while (current != null) {
            System.out.print(current.data + " ");
            current = current.next;
        }
        System.out.println();
    }
    public static void main(String[] args) {
        DLLInsertionAtBeginning dll= new DLLInsertionAtBeginning();
        
        dll.insertAtBeginningDLL(400);
        dll.insertAtBeginningDLL(300);
        dll.insertAtBeginningDLL(200);
        dll.insertAtBeginningDLL(100);
        
        System.out.println("Doubly Linked List after insertion:");
        dll.display();
    }
}